Sources/GRPCNIOTransportCore/Client/Connection/LoadBalancers/Subchannel.swift

```diff
@@ -646,9 +657,15 @@ extension Subchannel {
           self = .notConnected(NotConnected(from: state))
           onClosed = .emitIdle
 
-        case .keepaliveTimeout, .error(_, wasIdle: false):
+        case .keepaliveTimeout:
+          self = .notConnected(NotConnected(from: state))
+          onClosed = .emitTransientFailureAndReconnect(
+            cause: RPCError(code: .unavailable, message: "The keepalive timed out.")
```

Comment:
nit: I think we can add a little more detail to the message here

```suggestion
            cause: RPCError(code: .unavailable, message: "The connection became unresponsive and was closed because the keepalive timeout fired.")
```